girder/events.py

```diff
@@ -225,16 +227,17 @@ def bind(eventName, handlerName, handler):
     :type handler: function
     """
     if eventName in _deprecated:
-        girder.logger.warning('event "%s" is deprecated; %s' %
-                              (eventName, _deprecated[eventName]))
+        girder.logger.warning('event "%s" is deprecated; %s' % (eventName, _deprecated[eventName]))
 
     if eventName not in _mapping:
-        _mapping[eventName] = []
+        _mapping[eventName] = OrderedDict()
 
-    _mapping[eventName].append({
-        'name': handlerName,
-        'handler': handler
-    })
+    if handlerName in _mapping:
+        # We delete from the OrderedDict to move the handler to the end of the ordering
+        del _mapping[handlerName]
+        girder.logger.warning('Overwriting event binding: %s -> %s' % (eventName, handlerName))
```

Comment:
Doesn't this mean that a plugin could replace a core binding without intending to do so? Should subsequent calls to bind instead no-op and print the warning that they aren't being used?

Reply:
I'm fine with either behavior, I'll make it a no-op unless someone prefers it this way.

Reply:
After thinking about it some more, I actually prefer the behavior I had written at first, because it is more in line with how a `dict` works, and it also has additional power, in that it would allow plugin authors to intentionally overwrite event handlers from either core or their upstream plugins.